client/lib/signup/step-actions.js

```diff
@@ -28,6 +28,19 @@ import {
 import { getSiteTitle } from 'state/signup/steps/site-title/selectors';
 import { getSurveyVertical, getSurveySiteType } from 'state/signup/steps/survey/selectors';
 
+function createCart( callback, dependencies, data ) {
+	const { designType } = dependencies;
+	const { domainItem, themeItem } = data;
+
+	if ( designType === 'domain' ) {
+		// TODO: Find a way to create cart without a site
+		// SignupCart.addToCart depends on the site's slug
+		callback( undefined, [ null, null, domainItem, themeItem ] );
```

Comment:
@Tug @drewblaisdell I guess you know better what should be done here. I assume we need to build custom `/me/transactions` call which creates a site behind the scenes and adds domainItem to the cart.